NumericUI/scripts/mods/NumericUI/HudElementDodgeCount.lua

```diff
@@ -20,6 +20,59 @@ local color_efficient = Color.terminal_text_header(255, true)
 local color_inefficient = Color.ui_hud_warp_charge_low(255, true)
 local color_limit = Color.ui_hud_warp_charge_high(255, true)
 
+local timer_x_offset = (-UIWorkspaceSettings.screen.size[1]
++ scenegraph_definition.container.size[1])/2
+
+local timer_size_color = function(time_to_refresh, cooldown, current_dodges, force_show_max_width)
+	-- NB: time_to_refresh will increase towards zero
+	local t_max = cooldown
+	if (-time_to_refresh >= t_max or force_show_max_width) and mod:get("dodge_timer_hide_full") then
+		return
+	end
+	local natural_time = force_show_max_width and 0 or math.clamp((time_to_refresh + t_max) / t_max, 0, 1)
+	local timer_size = {
+		mod:get("dodge_timer_width") * (1 - natural_time),
+		mod:get("dodge_timer_height"),
+	}
+	local timer_color = {}
+	local color_start = Color[mod:get("color_start")](255, true)
+	local color_end = Color[mod:get("color_end")](255, true)
+	for i = 1, 4 do
+		table.insert(
+			timer_color,
+			math.lerp(
+				color_start[i],
+				color_end[i],
+				natural_time
+			)
+		)
+	end
+	-- Hide timer if dodges are full (useful when playing with the Agile blessing)
+	if current_dodges == 0 then
+		timer_color[1] = 0
+	end
+	return timer_size, timer_color
+end
+
+local timer_pass = 	{ {
+	style_id = "timer",
+	pass_type = "rect",
+	style = {
+		color = {255, 255, 100, 100},
+		vertical_alignment = "top",
+		horizontal_alignment = "center",
+		drop_shadow = true,
+		size = {100, 8},
+		offset = {
+			timer_x_offset,
+			mod:get("dodge_timer_y_offset"),
+		},
+	},
+	visibility_function = function(content, style)
+		return mod:get("dodge_timer")
+	end,
+} }
```

Comment:
if we're not using this for anything else i would just inline it in the definitions